highlighters/main/mh-parse.zsh

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+#
+# This file runs main highlighter on a specified file
+# i.e. parses the file with the highlighter. Outputs
+# running time (stderr) and resulting region_highlight
+# (stdout).
+#
+
+# Strive for -f
+# .. occuring problems (no q flag), but at least try to quote $0 and $@
+[[ -z "$ZSH_VERSION" ]] && exec /usr/bin/env zsh -f -c "source \"$0\" \"$1\" \"$2\" \"$3\""
+
+ZERO="${(%):-%N}"
+ZERO_RESOLVED="${ZERO:A:h}"
```

Comment:
You don't actually need $ZERO_RESOLVED; it'd be fine to replace all uses with $ZERO.

Yes, I know I said `${0:A:h}` originally.  I think I mislead you; sorry for that.